src/stirling/source_connectors/socket_tracer/socket_trace_connector.cc

```diff
@@ -384,18 +385,27 @@ auto SocketTraceConnector::InitPerfBufferSpecs() {
 
 Status SocketTraceConnector::InitBPF() {
   // PROTOCOL_LIST: Requires update on new protocols.
+  //
+  // The infer_protocol function does not work on kernels
+  // older than 5.2 when all protocols are enabled. Since
+  // the TraceMode::OnForNewerKernel enum results in a
+  // 'truthy' define, TraceMode::OnForNewerKernel and
+  // TraceMode::Off must be munged into 0 in order for the
+  // compiler to remove dead conditional statements resulting
+  // in a BPF program within the old program limit (< 4096 insns).
+  // This means the TransferSpec's enabled field must be used
+  // after SocketTraceConnector::EnableIfNeeded is called.
```

Comment:
First off, good catch!

Second: I feel like the piece about running out of BPF instructions should be move to where the `stirling_enable_<protocol>_tracing` flags are defined. There's more context there. Here we dutifully just use the protocol transfer specs to apply the rules.

```

```

Reply:
Sounds good. I've moved this above the first `stirling_enable_<protocol>_tracing` definition